packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -594,6 +594,7 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
     assert(hasPixels);
     assert(_pendingDimensions);
     activity!.applyNewDimensions();
+    context.didChangeContentDimensions();
```

Comment:
I'm just a little worried, because this api will be called in the layout stage, which will greatly restrict the application scenario, such as the tree cannot be rebuilt, and in the case of lazy loading, it will be called multiple times in the same frame.

Reply:
That is actually a good point, let me think about this more